File: programmers/level2/131127.js
// https://school.programmers.co.kr/learn/courses/30/lessons/131127
// 할인 행사
function solution(want, number, discount) {
  const wList = want.reduce((acc, cur, idx) => {
    acc[cur] = number[idx];
    return acc;
  }, {});
  // console.log(list)

  const total = (res) => Object.values(res).reduce((acc, cur) => acc + cur, 0);
  let answer = 0;
  // console.log('answer', answer);

  const dLen = discount.length;
  for (let i = 0; i < dLen; i++) {
    // console.log('i', i);
    const last = i + 10 > dLen ? dLen : i + 10;
    const list = { ...wList };
    for (let j = i; j < last; j++) {
      // console.log('discount[j]', discount[j]);
      if (list[discount[j]] > 0) {
        list[discount[j]] -= 1;
      }
    }
    const temp = total(list);
    // console.log('temp', temp);
    if (temp === 0) {
      answer += 1;
    }
  }
  return answer;
}

// console.log(
//   solution(
//     ['banana', 'apple', 'rice', 'pork', 'pot'],
//     [3, 2, 2, 2, 1],
//     [
//       'chicken',
//       'apple',
//       'apple',
//       'banana',
//       'rice',
//       'apple',
//       'pork',
//       'banana',
//       'pork',
//       'rice',
//       'pot',
//       'banana',
//       'apple',
//       'banana',
//     ],
//   ),
// ); //3
console.log(
  solution(
    ['apple'],
    [10],
    ['banana', 'banana', 'banana', 'banana', 'banana', 'banana', 'banana', 'banana', 'banana', 'banana'],
  ),
); //10

File: programmers/python/level2/131127.py
def solution(want, number, discount):
  want_dict = {key:number[idx] for idx, key in enumerate(want)}

  idx=0
  answer = 0
  while idx+10 < len(discount)+1:
    cart_dict={key:cnt for key,cnt in want_dict.items()}
    # print(cart_dict)
    # print('discount[idx:idx+10]',discount[idx:idx+10])
    for item in discount[idx:idx+10]:
      if item in want and cart_dict[item]>0:
        cart_dict[item] -= 1
    # print(cart_dict)
    if sum(cart_dict.values()) == 0:
      answer+=1
    idx+=1
  return answer

# print(solution(["banana", "apple", "rice", "pork", "pot"],	[3, 2, 2, 2, 1],	["chicken", "apple", "apple", "banana", "rice", "apple", "pork", "banana", "pork", "rice", "pot", "banana", "apple", "banana"]))
print(solution(["apple"],	[10],	["banana", "banana", "banana", "banana", "banana", "banana", "banana", "banana", "banana", "banana"]))
# ["banana", "apple", "rice", "pork", "pot"]	[3, 2, 2, 2, 1]	["chicken", "apple", "apple", "banana", "rice", "apple", "pork", "banana", "pork", "rice", "pot", "banana", "apple", "banana"]	3
# ["apple"]	[10]	["banana", "banana", "banana", "banana", "banana", "banana", "banana", "banana", "banana", "banana"] 0
